packages/mui-lab/src/SnackbarsProvider/SnackbarsProvider.tsx

```diff
@@ -0,0 +1,279 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Snackbar, { SnackbarProps } from '@mui/material/Snackbar';
+import Grow from '@mui/material/Grow';
+import SnackbarContent from '@mui/material/SnackbarContent';
+import { styled, useTheme } from '@mui/material/styles';
+import SnackbarsContext, { ShowSnackbarProps, SnackbarContentType } from './SnackbarsContext';
+import SnackbarsContainer from './SnackbarsContainer';
+
+const randomId = () => `mui-${Math.round(Math.random() * 1e5)}`;
+
+const StyledSnackbar = styled(Snackbar)<{ ownerState: ShowSnackbarProps }>(
+  ({ theme, ownerState }) => {
+    return {
+      [theme.breakpoints.up('sm')]: {
+        position: 'relative',
+        ...(ownerState.anchorOrigin!.horizontal !== 'center' && { left: 0, right: 0 }),
+        top: 0,
+        bottom: 0,
+        ...(ownerState.anchorOrigin!.vertical === 'bottom' && {
+          '&:not(:last-of-type)': {
+            marginTop: theme.spacing(),
+          },
+        }),
+        ...(ownerState.anchorOrigin!.vertical === 'top' && {
+          '&:not(:last-of-type)': {
+            marginBottom: theme.spacing(),
+          },
+        }),
+      },
+    };
+  },
+);
+
+export interface SnackbarsProviderProps
+  extends Omit<SnackbarProps, 'children' | 'classes' | 'key' | 'message' | 'onClose' | 'open'> {
+  /** The maximum number of snackbars to display at a time.
+   * @default 3
```

Comment:
Why have a limit by default? All the libraries I could benchmark don't have a limit by default

Reply:
I think it's reasonable to limit by default, otherwise it'd be a non-functional UX pattern? I don't feel like stacking a bunch of snackbars would be useful, it seems just bloated noise in the UI. 

Though I see that we could simply just recommend that instead of actually limiting from a component-level. 🤔

Reply:
Alright, I simply found it curious that all the alternatives don't have a limit but we do. I think that time will tell us if it works or not. We can simply stay attentive to the UX feedback we will get on the GitHub issues 👍

Reply:
`notistack` has default limit of 3 and `mantine` has 5. 

>  I don't feel like stacking a bunch of snackbars would be useful, it seems just bloated noise in the UI.

This is what came to my mind and thus added a limit after benchmarking.